src/FEValues/interface_values.jl

```diff
@@ -0,0 +1,343 @@
+# Defines InterfaceValues and common methods
+"""
+    InterfaceValues(grid::AbstractGrid, quad_rule::FaceQuadratureRule, func_interpol_a::Interpolation, [geom_interpol_a::Interpolation], [func_interpol_b::Interpolation], [geom_interpol_b::Interpolation])
+
+An `InterfaceValues` object facilitates the process of evaluating values, averages, jumps and gradients of shape functions
+and function on the interfaces of finite elements.
+
+**Arguments:**
+
+* `grid` : instance of the current grid.
+* `quad_rule_a`: an instance of a [`FaceQuadratureRule`](@ref) for element A.
+* `quad_rule_b`: an instance of a [`FaceQuadratureRule`](@ref) for element B.
+* `func_interpol_a`: an instance of an [`Interpolation`](@ref) used to interpolate the approximated function for element A.
+* `func_interpol_b`: an instance of an [`Interpolation`](@ref) used to interpolate the approximated function for element B.
+* `geom_interpol_a`: an optional instance of an [`Interpolation`](@ref) which is used to interpolate the geometry for element A.
+  It uses the default interpolation of the respective [`RefShape`](@ref) by default.
+* `geom_interpol_b`: an optional instance of an [`Interpolation`](@ref) which is used to interpolate the geometry for element B.
+  It uses the default interpolation of the respective [`RefShape`](@ref) by default.
+ 
+**associated methods:**
+
+* [`shape_value_average`](@ref)
+* [`shape_value_jump`](@ref)
+* [`shape_gradient_average`](@ref)
+* [`shape_gradient_jump`](@ref)
+
+**Common methods:**
+
+* [`reinit!`](@ref)
+* [`getnquadpoints`](@ref)
+* [`getdetJdV`](@ref)
+
+* [`shape_value`](@ref)
+* [`shape_gradient`](@ref)
+* [`shape_divergence`](@ref)
+* [`shape_curl`](@ref)
+
+* [`function_value`](@ref)
+* [`function_gradient`](@ref)
+* [`function_symmetric_gradient`](@ref)
+* [`function_divergence`](@ref)
+* [`function_curl`](@ref)
+* [`spatial_coordinate`](@ref)
+"""
+InterfaceValues
+
+struct InterfaceValues{IP_a, N_t_a, dNdx_t_a, dNdξ_t_a, T_a, dMdξ_t_a, QR_a, Normal_t_a, GIP_a,
+    IP_b, N_t_b, dNdx_t_b, dNdξ_t_b, T_b, dMdξ_t_b, QR_b, Normal_t_b, GIP_b,
+    dim, C} <: AbstractValues
+    face_values_a::FaceValues{IP_a, N_t_a, dNdx_t_a, dNdξ_t_a, T_a, dMdξ_t_a, QR_a, Normal_t_a, GIP_a}
+    face_values_b::FaceValues{IP_b, N_t_b, dNdx_t_b, dNdξ_t_b, T_b, dMdξ_t_b, QR_b, Normal_t_b, GIP_b}
+    # used for quadrature point syncing
+    grid::Grid{dim, C, Float64}
+    cell_a_idx::ScalarWrapper{Int}
+    cell_b_idx::ScalarWrapper{Int}
+    ioi::ScalarWrapper{InterfaceOrientationInfo}
```

Comment:
I am a bit confused why this is needed, an  in particular it would be nice not to have this.

Reply:
In particular, is it not possible to compute the mappings with just the quadrature rule? Assuming the rule is "symmetric". Is the problem that the relative rotation isn't known? Then perhaps `reinit!(iv::InterfaceValues, ic::InterfaceCache)` can do it? That has the node IDs so should be able to compute this.

Reply:
These would be moved to interface cache in the next PR

Reply:
Ah I see. Perhaps that should be merged first then?